### ISSUE TITLE
Redis-plus-plus version for package building purposes 

### ISSUE THREAD

**Issue by wingunder** [sewenew/redis-plus-plus#141]
Hi @sewenew,
I can now successfully build a proper Debian package, with a minor addition to the `CMakeLists.txt` file.

I might have missed searching for it, but I cannot find any version number related to the redis-plus-plus code.
If the version is not available, CPack defaults to version 0.1.1, which is probably not what we want.
Would it be possible for you to add a version number in some form so that it would be available in `CMakeLists.txt`?

BTW, once the above would be available, I'll make a PR with the Debian package generation stuff, if you're interested.
Thanks & regards

**Comment by sewenew** [comment]
Great! Thanks for the work!

I'm not familiar with that. Do I only need to specify the version as a parameter of `project()` function? For example:

```
# CMakeLists.txt
project(redis++ VERSION 1.2.1)
```

Regards

**Comment by wingunder** [comment]
Hi @sewenew,
I looked at what they do at: https://github.com/Tencent/rapidjson
Adapted, it'll look as follows, and works for me:
```
if(POLICY CMP0025)
    # detect Apple's Clang
    cmake_policy(SET CMP0025 NEW)
endif()
if(POLICY CMP0054)
    cmake_policy(SET CMP0054 NEW)
endif()

if (CMAKE_VERSION VERSION_LESS 3.0)
    project(redis++ CXX)
else()
    cmake_policy(SET CMP0048 NEW)
    project(redis++ VERSION 1.2.1 LANGUAGES CXX)
endif()
```
I'm still not up to speed with CMake, so I don't really get all of this CMP[0-9]+ stuff.
Anyway, it seems to work for RapidJson, and that's a pretty decent (user-base-size) project.

You can test it, by adding the following:
```
message("PROJECT_VERSION: ${PROJECT_VERSION}")
message("PROJECT_VERSION_MAJOR: ${PROJECT_VERSION_MAJOR}")
message("PROJECT_VERSION_MINOR: ${PROJECT_VERSION_MINOR}")
message("PROJECT_VERSION_PATCH: ${PROJECT_VERSION_PATCH}")
```
We could also put the version (1.2.1), into a variable and put it at the to of the file, so that it is easier to spot and change.
```
set(REDIS_PLUS_PLUS_VERSION 1.2.1)
.
.
    project(redis++ VERSION ${REDIS_PLUS_PLUS_VERSION} LANGUAGES CXX)
```
Hope this helps.
Regards

**Comment by sewenew** [comment]
Great! I’ll take a look at rapidjson’s solution, and give you feedback later.

Regards

**Comment by wingunder** [comment]
Thanks, no rush :+1:

**Comment by sewenew** [comment]
@wingunder Version info has been added to CMakeLists.txt. Please have a try.

Sorry for the delay, and thanks a lot for your work!

Regards

**Comment by wingunder** [comment]
Hi @sewenew,

Thanks a lot. This patch is exactly what I needed. It works for me :+1: 
I was planning to make a PR for the Debian package support, but I'm not there yet. I need to create a lib- and a dev-package, and that's a bit tricky with cmake, as far as I can see. Making a single deb package with everything in it, is simple, but this will never get accepted by Debian. My idea is to supply something that requires the absolute minimal work for the eventual package maintainer(s).
I'll keep you posted on my progress.

Please tell me if I should close this PR. You may also close it if you want to.
Regards

**Comment by sewenew** [comment]
You can keep it open until the Debian package support is done :) Thanks again!

Regards

**Comment by wingunder** [comment]
Hi  @sewenew 

I finally found some time to get a bit further down the line with this issue and I could submit a PR, #193.

Unfortunately I ran into several problems with cmake and could not manage to build a dev _and_ a pure lib package. For now it's just a dev package with the relevant stuff in it. Optimally, it should be 2 packages, but I failed to get to the right cmake-voodoo to achieve this. Maybe there are some cmake cracks that can manage this.  

At least a basic Debian package can now be built.

Regards
